crates/cheatnet/src/cheatcodes/deploy.rs

```diff
@@ -77,6 +78,19 @@ impl CheatnetState {
                     ))))
                 }),
             CallContractOutput::Panic { panic_data } => {
+                let err_str = as_cairo_short_string(&panic_data[0]).unwrap();
+                for invalid_calldata_msg in [
+                    "Failed to deserialize param #",
+                    "Input too long for arguments",
+                    "INVALID_CALLDATA_LEN",
```

Comment:
I can't see this error in the test. How to obtain it?

Reply:
`INVALID_CALLDATA_LEN` appears when no `constructor` exists in a contract and the calldata.len() != 0

We have a separate `if` for this case in the deploy cheatcode (https://github.com/foundry-rs/starknet-foundry/pull/619/files#diff-acb75fb617fd5f282a6613f7233f2efeb1adc8b67c4bcb4f709d863da752e92aR48-R51)

Reply:
So actually there is no need to include it here

Reply:
Hmm, now I wonder if we should check constructor existence then. Shouldn't we allow blockifier to do it for us?

Reply:
We can do that, this will simplify the logic making blockifier the only one responsible for the calldata validation